opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1807,5 +1544,250 @@ namespace Opm {
+
+        auto& well_state = well_state_;
+        const std::string from = well_state.currentProductionGroupControl(group.name());
+
+        std::ostringstream ss;
+        ss << "Group " << group.name() << " exceeding "
+           << from << " limit  \n";
+        switch(exceed_action) {
+        case GroupProductionExceedLimit::ActionEnum::NONE: {
+            OPM_DEFLOG_THROW(std::runtime_error, "Group " + group.name() + "GroupProductionExceedLimit NONE not implemented", deferred_logger);
+            break;
+        }
+        case GroupProductionExceedLimit::ActionEnum::CON: {
+            OPM_DEFLOG_THROW(std::runtime_error, "Group " + group.name() + "GroupProductionExceedLimit CON not implemented", deferred_logger);
+            break;
+        }
+        case GroupProductionExceedLimit::ActionEnum::CON_PLUS: {
+            OPM_DEFLOG_THROW(std::runtime_error, "Group " + group.name() + "GroupProductionExceedLimit CON_PLUS not implemented", deferred_logger);
+            break;
+        }
+        case GroupProductionExceedLimit::ActionEnum::WELL: {
+            OPM_DEFLOG_THROW(std::runtime_error, "Group " + group.name() + "GroupProductionExceedLimit WELL not implemented", deferred_logger);
+            break;
+        }
+        case GroupProductionExceedLimit::ActionEnum::PLUG: {
+            OPM_DEFLOG_THROW(std::runtime_error, "Group " + group.name() + "GroupProductionExceedLimit PLUG not implemented", deferred_logger);
+            break;
+        }
+        case GroupProductionExceedLimit::ActionEnum::RATE: {
+            well_state.setCurrentProductionGroupControl(group.name(), GroupProduction::ControlEnum2String(newControl));
+            ss << "Switching control mode for group to " << GroupProduction::ControlEnum2String(newControl)
+               << " \n Wells in group " + group.name() + " switches to GRUP control limit";
+            wellGroupHelpers::setGroupControl(group, schedule(), reportStepIdx, false, well_state);
+            break;
+        }
+        default:
+            throw("Unvalid procedure for maximum rate limit selected for group" + group.name());
+        }
+
+        auto cc = Dune::MPIHelper::getCollectiveCommunication();
+        if (cc.size() > 1) {
+            ss << " on rank " << cc.rank();
+        }
+        deferred_logger.info(ss.str());
+
+
+    }
+
+
+    template<typename TypeTag>
+    void
+    BlackoilWellModel<TypeTag>::
+    actionOnBrokenConstraints(const Group2& group, const GroupInjection::ControlEnum& newControl, const int reportStepIdx, Opm::DeferredLogger& deferred_logger) {
+        auto& well_state = well_state_;
+        const std::string from = well_state.currentProductionGroupControl(group.name());
+        std::ostringstream ss;
+        ss << "Group " << group.name() << " exceeding  \n"
```

Comment:
The message is good, but I suggest to change the formatting a little, both here and for the producers (in the function above). Current example:
```
Group A exceeding  
FLD limit Switching control mode for group to VREP 
 Wells in group A switches to GRUP control limit
```

Suggested change:
```
Group A exceeding FLD limit. Switching control mode for group to VREP.
Wells in group A will be switched to GRUP control limit.
```

Reply:
Already done.